src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -0,0 +1,126 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SolutionBindingInformationProvider.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using EnvDTE;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using SonarLint.VisualStudio.Integration.Binding;
+using SonarLint.VisualStudio.Integration.Persistence;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal class SolutionBindingInformationProvider : ISolutionBindingInformationProvider
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        public SolutionBindingInformationProvider(IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
+        }
+
+        #region ISolutionBindingInformationProvider
+        public IEnumerable<Project> GetBoundProjects()
+        {
+            var bindingSerializer = this.serviceProvider.GetService<ISolutionBindingSerializer>();
+            bindingSerializer.AssertLocalServiceIsNotNull();
```

Comment:
Just a thought, rather than having to `AssertLocalServiceIsNotNull` every time, why not just assert (on the `VsSessionHost`) when a service is requested but is not satisfied? Reduces the clutter from every consumer. If you've requested a service when it's because you want to use it.